internal/dag/gatewayapi_processor.go

```diff
@@ -106,11 +118,82 @@ func (p *GatewayAPIProcessor) Run(dag *DAG, source *KubernetesCache) {
 
 		// Process all the routes that match this Gateway.
 		for _, matchingRoute := range matchingRoutes {
-			p.computeHTTPRoute(matchingRoute)
+			p.computeHTTPRoute(matchingRoute, listenerSecret)
 		}
 	}
 }
 
+func (p *GatewayAPIProcessor) validGatewayTLS(listener gatewayapi_v1alpha1.Listener) *Secret {
+
+	// Validate the CertificateRef is configured.
+	if listener.TLS.CertificateRef == nil {
```

Comment:
Worth adding status tests for the various status conditions that might be set for errors?

Reply:
These are Gateway statuses which were going to be handled by the Operator. Since then we've discussed this happening inside Contour, but that is happening in a separate branch.

Reply:
ah right, forgot about that